shell/platform/fuchsia/flutter/vsync_waiter.cc

```diff
@@ -31,13 +34,30 @@ VsyncWaiter::VsyncWaiter(std::string debug_label,
     wait->Cancel();
 
     FireCallbackNow();
-  };
+
+    // Generate a WeakPtrFactory for use with the UI thread. This does not need
+    // to wait on a latch because we only ever use the WeakPtrFactory on the UI thread
+    // so we have ordering guarantees (see ::AwaitVSync())
+    fml::TaskRunner::RunNowOrPostTask(
+      task_runners_.GetUITaskRunner(), fml::MakeCopyable([this]() mutable {
+        this->weak_factory_ui_ =
+            std::make_unique<fml::WeakPtrFactory<VsyncWaiter>>(this);
+      }));
+    };
 
   session_wait_.set_handler(wait_handler);
 }
 
 VsyncWaiter::~VsyncWaiter() {
   session_wait_.Cancel();
+
+  fml::AutoResetWaitableEvent ui_latch;
+  fml::MakeCopyable([weak_factory_ui = std::move(weak_factory_ui_),
```

Comment:
Don't you need a |RunNowOrPostTask| here for the single threaded configuration?

Reply:
Oh good catch. I copied the code from the equivalent code in Shell's ctor/dtor. Do we need to change Shell as well?

Reply:
I'd think so. Not sure how this wasn't covered by test already. Maybe add an explicit single threaded configuration to the shell_unittests (in a later patch is fine)?

Reply:
Sorry, I'm not firing on all cylinders today. The Shell code does have it, I just accidentally omitted it.